doc/api/events.md

```diff
@@ -586,6 +586,53 @@ to indicate an unlimited number of listeners.
 
 Returns a reference to the `EventEmitter`, so that calls can be chained.
 
+### async emitter.when(eventName[, options])
+<!-- YAML
+added: REPLACEME
+-->
+
+* `eventName` {any} The name of the event.
+* `options` {Object}
+  * `prepend` {boolean} True to prepend the handler used to resolve the
+    `Promise` to the handler queue.
```

Comment:
Does this even make a difference since any handler is called asynchronously anyway?

Reply:
Likely not. This is also something I've gone back and forth on so I'm happy to pull this back out if it doesn't make sense

Reply:
FWIW, This *would* have an effect when registering multiple `when` Promises since those would be resolved in a specific order... e.g.

```js
// `a` will always be printed before `b`
ee.when('foo').then(() => console.log('b'));
ee.when('foo', { prepend: true }).then(() => console.log('a'));
ee.emit('foo');
```

```js
// `b` will always be printed before `a`
ee.when('foo').then(() => console.log('b'));
ee.when('foo', { prepend: false }).then(() => console.log('a'));
ee.emit('foo');
```

Reply:
> since those would be resolved in a specific order

Yes, and I would consider relying on promise resolution order a big anti-pattern ;)

Reply:
I certainly cannot disagree with that :-)